src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java

```diff
@@ -1019,8 +1082,29 @@ public final Response getEventOverviews(@Context final HttpServletRequest reques
                 if (!(c instanceof  IsaacEventPageDTO)) {
                     continue;
                 }
-
                 IsaacEventPageDTO e = (IsaacEventPageDTO) c;
+
+                // Event leaders can only see the events which they are associated with (through the event's group token)
+                if (currentUser.getRole() == Role.EVENT_LEADER) {
+                    try {
+                        if (e.getIsaacGroupToken() == null || e.getIsaacGroupToken().isEmpty()) {
+                            continue;
+                        }
+
+                        AssociationToken eventGroupToken =
+                                userAssociationManager.lookupTokenDetails(currentUser, e.getIsaacGroupToken());
+                        UserGroupDTO eventGroup = groupManager.getGroupById(eventGroupToken.getGroupId());
+
+                        if (!GroupManager.isOwnerOrAdditionalManager(eventGroup, currentUser.getId())) {
```

Comment:
I wonder if this needs to happen more often and as part of the association check? (i.e. previous methods I've seen further up the page)